File: back/DevC/src/main/java/com/back/devc/domain/post/post/dto/AdminPostDetailResponse.java
package com.back.devc.domain.post.post.dto;

import com.back.devc.domain.post.post.entity.Post;

import java.time.LocalDateTime;

public record AdminPostDetailResponse(
        Long postId,
        String title,
        String content,
        Long userId,
        String writerName,
        Long categoryId,
        int viewCount,
        int likeCount,
        int commentCount,
        boolean isDeleted,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        LocalDateTime deleteAt
) {
    public static AdminPostDetailResponse from(Post post) {
        return new AdminPostDetailResponse(
                post.getPostId(),
                post.getTitle(),
                post.getContent(),
                post.getMember() != null ? post.getMember().getUserId() : null,
                post.getMember() != null ? post.getMember().getNickname() : null,
                post.getCategory().getCategoryId(),
                post.getViewCount(),
                post.getLikeCount(),
                post.getCommentCount(),
                post.isDeleted(),
                post.getCreatedAt(),
                post.getUpdatedAt(),
                post.getDeletedAt()
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/post/dto/AdminPostDetailResponse.kt
package com.back.devc.domain.post.post.dto

import com.back.devc.domain.post.post.entity.Post
import java.time.LocalDateTime

data class AdminPostDetailResponse(
    val postId: Long,
    val title: String,
    val content: String,
    val userId: Long?,
    val writerName: String?,
    val categoryId: Long,
    val viewCount: Int,
    val likeCount: Int,
    val commentCount: Int,
    val isDeleted: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val deletedAt: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(post: Post): AdminPostDetailResponse {
            return AdminPostDetailResponse(
                postId = post.postId
                    ?: throw IllegalStateException("Post ID cannot be null"),
                title = post.title,
                content = post.content,
                userId = post.member?.userId,
                writerName = post.member?.nickname,
                categoryId = post.category.categoryId,
                viewCount = post.viewCount,
                likeCount = post.likeCount,
                commentCount = post.commentCount,
                isDeleted = post.isDeleted,
                createdAt = post.createdAt,
                updatedAt = post.updatedAt,
                deletedAt = post.deletedAt
            )
        }
    }
}